airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java

```diff
@@ -104,7 +111,18 @@ public AirbyteMessage next() {
             .withStream(stream.getStream().getName())
             .withNamespace(stream.getStream().getNamespace())
             .withEmittedAt(emittedAt.toEpochMilli())
-            .withData(jsonNode)
-        );
+            .withData(jsonNode));
+  }
+
+  private boolean shouldContinue() {
+    if (count > 0) {
+      if (batchSize > 0) {
+        return count % batchSize == 0;
+      } else {
+        return true;
+      }
+    } else {
+      return false;
+    }
```

Comment:
This isn't a `should continue` method but rather a `should emit state` method.  

This change introduces a bug where no state will be emitted if `batchSize = 0`.

With the previous implementation, the `batchSize` and the `emit state every N messages` were coupled together.  I wonder if we should decouple this now if we want to support a `batchSize = 0`.

Reply:
@colesnodgrass Thanks for the clarification.  I'm not really sure how to decouple it, other than creating a different implementation to select when the batch size is 0.  Any other suggestions?

Reply:
I think for now if `batchSize == 0` we should default it to some value (500? 1000?) and revisit this decision in the future.